app/helpers/admin_area_helper.rb

```diff
@@ -25,8 +25,15 @@ def html_editor_needed?
         %w[ posts pages templates ].include?( controller_name ) )
   end
 
-  def plugins_for_menu
-    ::Plugin.loaded.map( &:underscore )
+  def plugins_for_admin_menu
+    plugins = []
+    ::Plugin.loaded.each do |camelcase|
+      underscored = camelcase.underscore
+      if File.exist? Rails.root.join "plugins/#{camelcase}/app/views/#{underscored}/admin/menu/_section.html.erb"
+        plugins << underscored
+      end
+    end
+    plugins
```

Comment:
Perhaps `::Plugin.loaded` should return an array of `Plugin` or `ShinyPlugin` objects so the objects can have useful behaviour attached, to allow something like:

```rb
def plugins_for_admin_menu
  plugins = []
  ::Plugin.loaded.each do |plugin|
    plugins << plugin if plugin.in_admin_menu?
  end
  plugins.map(&:underscored_name)
end
```